LandSandBoat/server
# Lua, SQL: Implement Zipacna pathing and interaction with Cermet Doors

**_I affirm:_**

## What does this pull request do?
Currently, Zipacna only spawns in one place and paths through the doors regardless if they are open or closed. 
This PR addresses the following: (I made no modifications to Zipacna's abilities or stats)
- adds the 4 spawn points located as shown here. 
![image](https://github.com/user-attachments/assets/d9385703-103a-4d8a-95d6-3f5bc5cd3a24)
![image](https://github.com/user-attachments/assets/833a10fd-b3bd-4b09-8b42-36c9a87db275)
![image](https://github.com/user-attachments/assets/58f35a18-16f1-4345-9ae7-99b6331d1983)
https://www.youtube.com/watch?v=nWjGMWHsg9k

The doors are labelled as such (from west side to east side)
Basement Loop - |1st Blue Door | Middle Blue Door | Middle Yellow Door | 1st Yellow Door| - Basement loop

- There are 4 doors controlled by a Yellow or Blue Monolith. When one is activated the other "side doors" are closed. 
** *Note - It is possible that Zipacna can get stuck between the middle yellow and first yellow door as well as the middle blue door and first blue door. If someone switches while he is pathing between**

Zipacna's behavior is to path to a door. If the door is closed, it will turn around and go to the next door in the sequence.
Example: Middle Yellow door was closed, Zipacana will turn around and go to the middle blue door. If the middle blue door is open, Zipacna will proceed to the 1st Blue door, make a loop in the basement and then return to the first blue door. If however, the yellow switch is then triggered, the middle blue door is now closed and once Zipacna arrives will again turn around and go to the middle yellow door. 

Due to there needing to be checks on both sides of the door, vars were used to track the last door Zip passed through. It is hard to follow but the video shows an hour and a half of Zipacna pathing with all the scenarios I could come up with that could happen. https://youtu.be/rWGuV6pqJww

TODO: I believe it is more of a navmesh thing but if he is engaged and the players wipe/zone Zipacna will likely path through walls back to its spawn point, at which point, will begin his pathing again. So suck less?




## Steps to test these changes
!spawnmob 17502570 (Will spawn in 1 of 4 locations.
!gotoid 17502570 - follow Zipacna and you will observe it path to each sides respective first doors. At that point, the pathing between doors takes over.

I understand that there is most likely not a way to clean up the path points, but if there is a way to clean up the door logic, I am open to suggestions.
